xcm/pallet-xcm/src/lib.rs

```diff
@@ -1095,6 +1095,62 @@ pub mod pallet {
 /// The maximum number of distinct assets allowed to be transferred in a single helper extrinsic.
 const MAX_ASSETS_FOR_TRANSFER: usize = 2;
 
+impl<T: Config> XcmQueryHandler for Pallet<T> {
+	type QueryId = u64;
+	type BlockNumber = T::BlockNumber;
+
+	/// Consume `message` and return another which is equivalent to it except that it reports
+	/// back the outcome.
+	///
+	/// - `message`: The message whose outcome should be reported.
+	/// - `responder`: The origin from which a response should be expected.
+	/// - `timeout`: The block number after which it is permissible for `notify` not to be
+	///   called even if a response is received.
+	///
+	/// `report_outcome` may return an error if the `responder` is not invertible.
+	///
+	/// It is assumed that the querier of the response will be `Here`.
+	///
+	/// To check the status of the query, use `fn query()` passing the resultant `QueryId`
+	/// value.
```

Comment:
This needs to be documented on the trait function, not on the implementation of the trait function.